Core/src/org/sleuthkit/autopsy/guicomponeontutils/JFileChooserHelper.java

```diff
@@ -0,0 +1,87 @@
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.guicomponeontutils;
+
+import java.awt.Cursor;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.FutureTask;
+import javax.swing.JFileChooser;
+import org.openide.windows.WindowManager;
+
+/**
+ * Helper class that will initialize JFileChooser instances in background threads.
+ * 
+ * On windows JFileChooser can take a long time to initialize. This helper class
+ * takes the work of initializing the JFileChooser off of the EDT so that in 
+ * theory the when the chooser is needed initialized and ready to be displayed 
+ * to the user.
+ * 
+ * https://stackoverflow.com/questions/49792375/jfilechooser-is-very-slow-when-using-windows-look-and-feel
+ */
+public final class JFileChooserHelper {
+    private final FutureTask<JFileChooser> futureFileChooser = new FutureTask<>(JFileChooser::new);
+    private JFileChooser chooser;
+    
+    /**
+     * Get a helper instance.
+     * 
+     * @return 
+     */
+    public static JFileChooserHelper getHelper() {
+        return new JFileChooserHelper();
+    }
+
+    /**
+     * Create a new instance of the helper class. The constructor will 
+     * kick of an executor to that will execute the task of initializing the 
+     * JFileChooser.
+     */
+    private JFileChooserHelper() {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        executor.execute(futureFileChooser);
+    }
+
+    /**
+     * Return and instance of JFileChooser to the caller.
+     * 
+     * This call may block the EDT if the JFileChooser initialization has not
+     * completed. 
+     * 
+     * @return 
+     */
+    public JFileChooser getChooser() {
+        if (chooser == null) {
+            // In case this takes a moment show the wait cursor.
+            WindowManager.getDefault().getMainWindow().setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR));
+            try {
+                // get() will only return when the initilization of the 
+                // JFileChooser has completed.
+                chooser = futureFileChooser.get();
+            } catch (InterruptedException | ExecutionException ex) {
+                // If an exception occured create a new instance of JFileChooser.
+                chooser = new JFileChooser();
```

Comment:
I don't expect a failure to happen here.

I will add a comment.

Reply:
I need to revise the above statement after finally getting my implementation of "I'm wondering if we can avoid this code duplication by passing a class to the chooser factory and calling newInstance()." to work.

Generally speaking I do not expect an exception to occur when initializing a JFileChooser, however I have found if you are subclassing JFileChooser as the code does in ExtractUnallocAction and HashDbCreateDatabaseDialog that you do not setup the subclass of JFileChooser correctly that you will get a reflection exception when using the Factory.

In other words if you create a subclass of JFileChooser and make the default constructor private\protected then JFileChooserFactory will not be able to find the constructor and you would get an exception when get is called.

I am a little indecisive about how to handle the exception here now because as far as I can see an exception would only occur due a coding error.
1. getChooser could just throw an exception and let the calling class decided what to do.  Part of the inspiration for this class was to avoid duplicate code.  Now every where that we call getChooser would need a try catch.
2. log the exception, but still return an instance of JFileChooser. Given that this would occur when the code was trying initialize a custom version of JFileChooser this seem like it has the potential for unknown consequences. 
3. log exception, popup and error, have the method return null. Not much different than 2, I guess.
4. Something else I haven't thought of.